.changeset/thin-rice-promise.md

```diff
@@ -0,0 +1,9 @@
+---
+"@pnpm/plugin-commands-script-runners": minor
+"@pnpm/config": minor
+"pnpm": minor
+"@pnpm/plugin-commands-outdated": patch
+"@pnpm/plugin-commands-store": patch
+---
+
+Added cache for `pnpm dlx` [#5277](https://github.com/pnpm/pnpm/issues/5277).
```

Comment:
Where is it cached and for how long?

Reply:
5 minutes by default. Add `dlx-cache-max-age=N` to `.npmrc` to set it to N minutes.

Reply:
No, I changed it to 1 day.